OMEdit/OMEditLIB/Plotting/PlotWindowContainer.cpp

```diff
@@ -434,6 +443,7 @@ void PlotWindowContainer::addArrayParametricPlotWindow()
                                          pPlottingPage->getHorizontalAxisTitleFontSizeSpinBox()->value(), pPlottingPage->getHorizontalAxisNumbersFontSizeSpinBox()->value(), pPlottingPage->getFooterFontSizeSpinBox()->value(),
                                          pPlottingPage->getLegendFontSizeSpinBox()->value());
     addCloseActionsToSubWindowSystemMenu(pSubWindow);
+    addRenameTabToSubWindowSystemMenu(pSubWindow);
```

Comment:
@adeas31 

I did this now for the animation window. However, when I attempted it for the diagram window, I got a segmentation fault while testing, possibly due to its being singular. Would it be ok to leave it out for this first version of the feature? If not, do you have any idea what might cause the segmentation failure?

Reply:
Yeah the mdiarea contains 3 types of windows i.e, plot, animation and diagram. You are calling `getCurrentWindow()` inside `renamePlotWindow()` which will give you active PlotWindow or null pointer. You need to change your code since its not necessary that the action is invoked from the active window. You should instead do this,

```
QAction *pAction = qobject_cast<QAction*>(sender());
QMdiSubWindow *pMdiSubWindow = qobject_cast<QMdiSubWindow*>(pAction->parent());
```

And create the QAction inside `addRenameTabToSubWindowSystemMenu()` like this,

```
QMenu *pMenu = pMdiSubWindow->systemMenu();
QAction *pRenamePlotWindowAction = new QAction(tr("Rename"), pMdiSubWindow);
pRenamePlotWindowAction->setStatusTip(tr("Renames the plot tab"));
connect(pRenamePlotWindowAction, SIGNAL(triggered()), SLOT(renamePlotWindow()));
pMenu->addAction(pRenamePlotWindowAction);
```

This will guarantee that you know exactly for which subwindow the menu is invoked.

Now I also realized that `Close All Windows But This` implementation is also wrong. It is also checking for active window instead of checking from which window the menu item is invoked.